bdns/dns_test.go

```diff
@@ -190,13 +188,30 @@ func serveLoopResolver(stopChan chan bool) chan bool {
 			fmt.Println(err)
 		}
 	}()
-	return waitChan
+}
+
+func pollServer() {
+	var conn *dns.Conn
+	timeout := time.Duration(time.Millisecond * 200)
+	tries := 25
+
+	for i := 0; conn == nil; i++ {
+		if i == tries {
+			fmt.Println("Max tries reached")
+			os.Exit(1)
+		}
+		conn, _ = dns.DialTimeout("tcp", dnsLoopbackAddr, timeout)
+		if conn == nil {
```

Comment:
This should probably be `conn != nil` and break the loop instead of sleeping, the the `sleep` can be put on the outside of the loop below this `if`.